go/vt/mysqlctl/mysqld.go

```diff
@@ -586,6 +576,25 @@ func (mysqld *Mysqld) initConfig(root string) error {
 	return ioutil.WriteFile(mysqld.config.path, []byte(configData), 0664)
 }
 
+func getMycnfTemplates(root string) []string {
+	if mycnfTemplateFile != nil {
```

Comment:
I think this condition should be *mycnfTemplateFile != "" to not break the old default behavior (see the broken tests).